json_schema/module/project/publication.json

```
@@ -54,6 +54,13 @@
             "type": "string",
             "user_friendly": "Publication URL",
             "example": "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5667944/"
+        },
+        "is_hca_publication": {
```

is it likely/possible that a project has more than 1 publication?

---

Yes, it is possible, although if I have to go by gut intuition, not very likely. This module gets loaded in 

https://github.com/HumanCellAtlas/metadata-schema/blob/c1e477d50f64c65339fa1f8c83b29aff2a753aed/json_schema/type/project/project.json#L61-L67

as an array, so multiple publications is possible. I can't remember if there is currently any case in the DCP but we have recorded instances of the same project in more than one publication in our backoffice.

---

awesome -- so there's two different things happening here -- one which is the property name implies the response is true/false (has_*) and then we want an array list to consume hca publications. I might propose that we just call this property "hca_publication" (so the boolean assumption no longer applies) and then the array can be 0 or n elements

---

The name is a bit awkward. "publication" is implied by the context. There is no precedent for an "is_" prefix in the name of other boolean properties:

https://github.com/HumanCellAtlas/metadata-schema/search?l=JSON&q=boolean

Can I throw 

```suggestion
        "part_of_hca": {
```

in the ring? This is based on the wording on https://www.humancellatlas.org/publications/

---

Alternatively, we could model this more richly by making this an enum that tracks the review process:

```
"hca_review": {
    "type": "string",
    "enum": [ "pending", "rejected", "accepted" ]
```

---

>I think there's a misunderstanding here. This schema describes a single publication. The project schema has a property that is a list of documents that match this schema. Any number of the project's publications can be an official HCA publication. I think this is modeled correctly.

Yes, spot on! The field itself is not an array, but the property in the `project` type schema that loads the module is.

About the name of the field, I've been discussing with the team, and `part_of_hca` does not sound less confusing to some members, as some members pointed out everything we ingest will be part of HCA, even if it's not an official hca publication. How about we drop the `is` for an `official`? (`official_hca_publication`?)

I understand there is a concern about redundancy, but that's how they are officially named, hca publications. Happy to jump on a call to discuss further about this!

---

>Alternatively, we could model this more richly by making this an enum that tracks the review process:
>.
>.


About the last comment, I like the idea to make the model more rich, but I think it will make the process very complicated. We usually only have access to the final decision if it's positive, so getting (and updating over time when it changes!) the status may make it a bit complicated.

---

@ESapenaVentura Yes`official_hca_publication` seems fine. Also would be great to make this required if possible. 

I presume that making this field optional for now makes it possible to do a simpler data migration and mark the official HCA papers only. Is this correct? Or is it about the same effort to then mark all non HCA publications "false" in the same go? 

My preference would be to make it required now if it does not create an undue burden on the data ops to backfill. 

My understanding (and correct me if I am wrong) is that the metadata is "Open world" meaning that the absence of a fact does not imply the fact is false, but only that it is unknown by the system.

---

Perfect, thanks! `official_hca_publication` it is then!

>I presume that making this field optional for now makes it possible to do a simpler data migration and mark the official HCA papers only. Is this correct? Or is it about the same effort to then mark all non HCA publications "false" in the same go?

Whenever we do a data migration, we do have (in the tracking spreadsheet) a field to indicate papers that have been accepted as HCA publications, so (even if it's not super thorough) we may already have something to mark papers as hca publications!

>My preference would be to make it required now if it does not create an undue burden on the data ops to backfill.

I changed it yesterday, it is required now! I think the biggest burden will be how to "massively" migrate the project json, but I think we have the tools and it's just a matter of having someone on the team work on it.

>My understanding (and correct me if I am wrong) is that the metadata is "Open world" meaning that the absence of a fact does not imply the fact is false, but only that it is unknown by the system.

💯  spot on!